### PR TITLE
[DT-131][risk=no] Fix dataset bug with fitbit sleep concept sets

### PR DESCRIPTION
**Pull request by dolbeew**

### CODE REVIEW

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 1]
```diff
@@ -746,6 +746,14 @@ const apiEnumToPrePackageConceptSets = (
...
         break;
       }
+      case PrePackagedConceptSetEnum.FITBITSLEEPDAILYSUMMARY: {
+        re.add(PrepackagedConceptSet.FITBITSLEEPDAILYSUMMARY);
+        break;
+      }
+      case PrePackagedConceptSetEnum.FITBITSLEEPLEVEL: {
+        re.add(PrepackagedConceptSet.FITBITSLEEPLEVEL);
+        break;
+      }
```
Comment (freemabd) [inline ui/src/app/pages/data/data-set/dataset-page.tsx:756]: Can we make this more bullet proof? You should be able to do something like this so this doesn't bite us everytime we add a new prepackaged type:
```suggestion
      case PrePackagedConceptSetEnum.BOTH: {
        re.add(PrepackagedConceptSet.PERSON);
        re.add(PrepackagedConceptSet.SURVEYS);
        break;
      }
      case PrePackagedConceptSetEnum.SURVEY: {
        re.add(PrepackagedConceptSet.SURVEYS);
        break;
      }
      case PrePackagedConceptSetEnum.NONE: {
        break;
      }
      default: {
        re.add(PrepackagedConceptSet[pre]);
        break;
      }
```

**Review by freemabd — COMMENTED** [review]
Added a comment about making this more robust.

File: ui/src/app/pages/data/data-set/dataset-page.tsx (hunk 1 above)
Comment (dolbeew) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: Yeah, good call. I had to add a reverse enum to get the correct value but definitely cleaner and better going forward this way.

Comment (dolbeew) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: Realized we basically had the same issue of needing to update with the static reverse enum. Just pushed a change to build the reverse enum based on the current enum properties.

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 2]
```diff
@@ -1372,6 +1356,16 @@ export const DatasetPage = fp.flow(
...
+            case PrepackagedConceptSet.FITBITSLEEPDAILYSUMMARY:
+              selectedPrePackagedConceptSDetEnum.push(
+                PrePackagedConceptSetEnum.FITBITSLEEPDAILYSUMMARY
+              );
+              break;
+            case PrepackagedConceptSet.FITBITSLEEPLEVEL:
+              selectedPrePackagedConceptSDetEnum.push(
+                PrePackagedConceptSetEnum.FITBITSLEEPLEVEL
+              );
+              break;
```
Comment (freemabd) [inline ui/src/app/pages/data/data-set/dataset-page.tsx:1368]: Can we also apply this pattern to this switch statement as well?

Comment (dolbeew) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: Couldn't get the same conversion to work for this enum, kept complaining about the types. After trying for a while, I realized both conversions don't seem necessary, so I changed all handling of prepackaged sets to use `PrePackagedConceptSetEnum` and deleted `PrepackagedConceptSet` and all conversion functions. Seems to be working properly, if you can give it another run through to verify.

**Pull request merged**